Feature: Update Message Notification

mazma1 opened the pull request:
What does this PR do?
* Add model validations
* Update model field names from snake_case to camelCase
* version API routes
* Refactor client and server to reflect changes in model field names and API endpoints
* Display full message content on a different component
* Updates unread message notification when a message is opened
* Displays group messages and members using the group id param passed in URL

mazma1 committed:
refactor: implement defense feedback
- Refactor search feature to use a search query
- Add version(v1) to API routes
- Handle errors for API requests

mazma1 committed:
refactor: implement defense feedback
    - Model validations
    - Change model field names to camelCase

mazma1 committed:
Commit before mentorship session

mazma1 committed:
refactor: implement defense feedback
 - Refactor client to reflect model changes

mazma1 committed:
feature(refactor):
- Render full message on a different URL
- Unread notifications get updated (reduced) when the message component mounts

mazma1 committed:
feature(refactor):
- First group on a message board is not being set as active by default any longer
- Persists active group by passing the group id in the URL

mazma1 committed:
feature(refactor):
- Add a dashboard component to display message when there is no active group
- Display search button only when there is an active group

gbolly commented on app/client/components/client-frame/Header.jsx:
```diff
@@ -7,11 +7,11 @@ import toastr from 'toastr';
 import $ from 'jquery';
```
Not sure if you are using this anywhere but generally it is a bad practice to mix jquery with react. You'll have to be extra careful with the DOM states and performance issues if you want to mix both, and it's sometimes much easier not to use jQuery since react does a great job in handling events.

gbolly commented on app/client/components/message-board/MessageBody.jsx:
```diff
@@ -9,93 +13,180 @@ import {
+    this.props.updateReadStatus(messageParams);
+  }
+
+  onClick() {
```
Giving this function a more descriptive function name would make the code more readable. Also, could you extract `localStorage.getItem('groupId')` into a const. That way, the code also looks readable and understandable. So, the link can read `/message-board/${groupId}`

gbolly commented on app/server/controllers/user.js:
```diff
@@ -188,23 +190,23 @@ export default {
             <br><br>You recently made a request to reset your Post It password. 
             Please click the link below to complete the process. 
-            <br><br><a href='http://${req.headers.host}/newpassword/${resetPasswordHash}'>Reset now ></a>
+            <br><br><a href='http://${req.headers.host}/new-password/${resetPasswordHash}'>Reset now ></a>
```
Curious to know if this works.

mazma1 replied:
Yes, it works.

gbolly reviewed (commented):
Great job. This looks good. Just a few changes and you're g2g.

mazma1 committed:
Initial commit for pagination

mazma1 committed:
feature(refactor):

- Finish pagination
- Fetch group messages after updating read message

mazma1 committed:
chore(refactor):
- Remove unnecessary rules from .eslintrc
- Add .hound.yml file to use .eslintrc for linting files on repo
-

mazma1 committed:
Add .jshintrc to specify es6 version to hound
